Python/optimizer_bytecodes.c

```diff
@@ -362,6 +362,7 @@ dummy_func(void) {
     }
 
     op(_TO_BOOL_NONE, (value -- res)) {
+        REPLACE_OPCODE_IF_EVALUATES_PURE(value);
```

Comment:
I think this already gets replaced in all of the cases we care about, so adding this probably isn't needed (unless you can find a case we're missing).